examples/fabric-admin/device_manager/DeviceSubscriptionManager.h

```diff
@@ -0,0 +1,88 @@
+
+class DeviceSubscriptionManager
+{
+public:
+    /// Attribute subscription to attributes that are important to keep track and send to fabric-bridge
+    /// via RPC when change has been identified.
+    ///
+    /// An instance of DeviceSubscription is intended to be used only once. Once a DeviceSubscription is
+    /// terminal, either from an error or from subscriptions getting shut down, we expect the instance
+    /// to be deleted. Any new subscription should instantiate another instance of DeviceSubscription.
+    ///
+    /// class is public because OnDeviceConnectedWrapper and OnDeviceConnectionFailureWrapper need
+    /// cast back to DeviceSubscription.
+    class DeviceSubscription : public chip::app::ReadClient::Callback
+    {
+    public:
+        DeviceSubscription();
+
+        CHIP_ERROR StartSubscription(DeviceSubscriptionManager * manager, chip::Controller::DeviceController & controller,
+                                     chip::NodeId nodeId);
+
+        ///////////////////////////////////////////////////////////////
+        // ReadClient::Callback implementation
+        ///////////////////////////////////////////////////////////////
+        void OnAttributeData(const chip::app::ConcreteDataAttributePath & path, chip::TLV::TLVReader * data,
+                             const chip::app::StatusIB & status) override;
+        void OnReportEnd() override;
+        void OnError(CHIP_ERROR error) override;
+        void OnDone(chip::app::ReadClient * apReadClient) override;
+
+        ///////////////////////////////////////////////////////////////
+        // callbacks for CASE session establishment
+        ///////////////////////////////////////////////////////////////
+        void OnDeviceConnected(chip::Messaging::ExchangeManager & exchangeMgr, const chip::SessionHandle & sessionHandle);
+        void OnDeviceConnectionFailure(const chip::ScopedNodeId & peerId, CHIP_ERROR error);
+
+    private:
+        DeviceSubscriptionManager * mManager;
+        std::unique_ptr<chip::app::ReadClient> mClient;
+
+        chip::Callback::Callback<chip::OnDeviceConnected> mOnDeviceConnectedCallback;
+        chip::Callback::Callback<chip::OnDeviceConnectionFailure> mOnDeviceConnectionFailureCallback;
+
+        chip_rpc_AdministratorCommissioningChanged mCurrentAdministratorCommissioningAttributes;
+        bool mChangeDetected = false;
+        // Ensures that DeviceSubscription starts a subscription only once.  If instance of
+        // DeviceSubscription  can be reused, the class documentation should be updated accordingly.
+        bool mSubscriptionStarted = false;
+    };
+
+    static DeviceSubscriptionManager & Instance();
+
+    /// Usually called after we have added a synchronized device to fabric-bridge to monitor
+    /// for any changes that need to be propgated to fabric-bridge.
+    CHIP_ERROR StartSubscription(chip::Controller::DeviceController & controller, chip::NodeId nodeId);
+
+private:
```

Comment:
There's no apparent thread synchronization mechanism.  access to the mDeviceSubscriptionMap should be protected to prevent race conditions.

```
private:
    std::mutex mMapMutex;
```

Reply:
Offline discussed see https://github.com/project-chip/connectedhomeip/pull/35305/files#r1739096746